build.gradle

```diff
@@ -224,10 +224,51 @@ configurations.all {
         force "io.netty:netty-common:${versions.netty}"
         force "io.netty:netty-handler:${versions.netty}"
         force "io.netty:netty-transport:${versions.netty}"
-        force "io.netty:netty-transport-native-unix-common:${versions.netty}"
+        }
+}
+
+//create source set 'newTest'
+//add classes from the main source set to the compilation and runtime classpaths of the newTest
+sourceSets {
+    newTest {
+        compileClasspath += sourceSets.main.output
+        runtimeClasspath += sourceSets.main.output
     }
 }
 
+//add new task that runs new tests
+task newTest(type: Test) {
+    description = 'Run new tests.'
+    group = 'verification'
+    testClassesDirs = sourceSets.newTest.output.classesDirs
+    classpath = sourceSets.newTest.runtimeClasspath
+
+    //run the newTest task after the test task
+    shouldRunAfter test
+}
+
+//run the newTest task before the check task
+check.dependsOn newTest
+
+configurations {
```

Comment:
This seems to duplicate configuration at line 213, do we need both?

Reply:
It has been fixed. There is just one `configurations` declaration right now.